Source/Relativity.Test.Helpers/Relativity.Test.Helpers/ArtifactHelpers/ClientHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using Relativity.Services.Choice;
+using Relativity.Services.Client;
+using Relativity.Test.Helpers.ArtifactHelpers.Interfaces;
+
+namespace Relativity.Test.Helpers.ArtifactHelpers
+{
+	public class ClientHelper
+	{
+		public static int CreateClient(Services.ServiceProxy.ServiceFactory serviceFactory, string name)
+		{
+			try
+			{
+				using (IClientManager proxy = serviceFactory.CreateProxy<IClientManager>())
+				{
+					List<ChoiceRef> choiceRefs = proxy.GetStatusChoicesForClientAsync().Result;
+					ChoiceRef statusRef = choiceRefs.Find(x => x.Name == "Active");
+
+					var newClient = new Services.Client.Client
+					{
+						Name = name,
+						Number = Guid.NewGuid().ToString(),
+						Status = statusRef, Keywords = "Test Client",
+						Notes = "Created with Relativity Test Helpers."
+					};
+
+					var clientArtifactId = proxy.CreateSingleAsync(newClient).Result;

```

Comment:
We should not be using `.Result` on async calls. [Here's a write-up ](https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html) that will go into more detail.

In general, we should be using async / await all the way through our application's call stack. If we absolutely cannot, then we need to use `.ConfigureAwait(false).GetResult()`

Reply:
This comment applies to all locations where we're using `.Result`

Reply:
@relthorbiedrzycki I've actually brought this up before, but to do this correctly (using async and await through the application's call stack) would require a large rewrite of TestHelpers. I've replaced .Result with your suggestion.

Reply:
@jake-martin understood, thank you. Also, for some more reading in the case we have this discussion again in the future: https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#async-all-the-way

Reply:
Oh I've read this, I know this is not good 🙂